zulipterminal/ui_tools/boxes.py

```diff
@@ -144,13 +144,13 @@ def generic_autocomplete(self, text: str, state: Optional[int]
         if (state is not None and state < len(fewer_typeaheads)
                 and state >= -len(fewer_typeaheads)):
             typeahead = fewer_typeaheads[state]  # type: Optional[str]
-            if typeahead:
-                typeahead = text[:prefix_index] + typeahead
         else:
             typeahead = None
             state = None
         self.view.set_typeahead_footer(reduced_suggestions,
                                        state, is_truncated)
+        if typeahead:
+            typeahead = text[:prefix_index] + typeahead
```

Comment:
Is this change necessary, given the next commit? Aren't you moving it back?

Reply:
The git diff is a bit weird so it looks a little strange, but this statement stays in the same method.